servicetalk-grpc-netty/src/main/java/io/servicetalk/grpc/netty/DefaultGrpcServerBuilder.java

```diff
@@ -146,6 +146,8 @@ private ExecutionContextInterceptorHttpServerBuilder preBuild() {
         final ExecutionContextInterceptorHttpServerBuilder interceptor =
                 new ExecutionContextInterceptorHttpServerBuilder(httpServerBuilderSupplier.get());
 
+        interceptor.appendServiceFilter(GrpcTrailersOptimizationServiceFilter.INSTANCE);
```

Comment:
Having this filter at the very end, as close as possible to the `GrpcRouter`, will help to minimize the impact: makes sure there is no user code in between that can modify the response before aggregation and allows users to do their modifications after. Consider moving this line after invoking `initializer.initialize(interceptor);`